### PR TITLE
Add forward native function and restrict send/forward methods

### PR DESCRIPTION
**Pull request by chamil321**
This PR will add following changes. 
- Add "forward" native function to the response struct. 
- Restrict send/forward methods only to the outbound response. 
- Add test cases.
- Refactor samples

### CODE REVIEW

File: modules/ballerina-http/src/main/java/org/ballerinalang/net/http/Constants.java [hunk 1]
```diff
@@ -131,6 +131,7 @@
     public static final String RESPONSE = "Response";
     public static final String TYPE_STRING = "string";
     public static final String TRANSPORT_MESSAGE = "transport_message";
+    public static final String RESPONSE_STATUS = "response_status";
```
Comment (shafreenAnfar) [inline modules/ballerina-http/src/main/java/org/ballerinalang/net/http/Constants.java]: shall we at the word outbound to this ?

File: modules/ballerina-http/src/main/java/org/ballerinalang/net/http/HttpUtil.java [hunk 2]
```diff
@@ -691,4 +695,18 @@ public static String sanitizeUri(String uri) {
         }
         return "/".concat(uri);
     }
+
+    public static void methodInvocationCheck(BStruct bStruct) {
+        if (bStruct.getNativeData(METHOD_ACCESSED) != null) {
+            throw new BallerinaException("illegal function invocation");
```
Comment (shafreenAnfar) [inline modules/ballerina-http/src/main/java/org/ballerinalang/net/http/HttpUtil.java]: I guess saying illegal state is better. Shall we check with java.

Comment (chamil321) [reply modules/ballerina-http/src/main/java/org/ballerinalang/net/http/HttpUtil.java]: IllegalStateException is used when a method has been invoked at an illegal or inappropriate time. So it is applicable. Made the change

File: modules/ballerina-http/src/main/java/org/ballerinalang/net/http/Constants.java (hunk 1 above)
Comment (chamil321) [reply modules/ballerina-http/src/main/java/org/ballerinalang/net/http/Constants.java]: Changed to OUTBOUND_RESPONSE

**Pull request merged**